doc/server-secrets.md

```diff
@@ -0,0 +1,106 @@
+# Server Secrets
+
+It is possible to provide a token or credentials for a number of external
+services. These may be used to lift a rate limit or provide access to
+private resources from a self-hosted instance.
+
+Secrets can be set in  `private/secret.json`. For example:
+
+```
+{
+  "gh_token": "..."
+}
+```
+
+## Bintray
+
+* `bintray_user`
+* `bintray_apikey`
+
+The bintray API [requires authentication](https://bintray.com/docs/api/#_authentication)
+Create an account and obtain a token from the user profile page.
+
+
+## GitHub
+
+* `gh_token`
+
+Because of Github rate limits, you will need to provide a token, or else badges
+will stop working once you hit 60 requests per hour, the
+[unauthenticated rate limit][github rate limit].
+
+You can [create a personal access token][personal access tokens] through the
+Github website. When you create the token, you can choose to give read access
+to your repositories. If you do that, your self-hosted Shields installation
+will have access to your private repositories.
+
+When a `gh_token` is specified, it is used in place of the Shields token
+rotation logic.
+
+[github rate limit]: https://developer.github.com/v3/#rate-limiting
+[personal access tokens]: https://github.com/settings/tokens
+
+* `gh_client_id`
+* `gh_client_secret`
+
+## Jenkins CI
+
+* `jenkins_user`
+* `jenkins_pass`
+
+Provide a username and password to give your self-hosted Shields installation
+access to a private Jenkins CI instance.
+
+## JIRA
+
+* `jira_user`
+* `jira_pass`
+
+Provide a username and password to give your self-hosted Shields installation
+access to a private JIRA instance.
+
+For legacy reasons `jira_username` and `jira_password` are also supported
+but may be removed in future.
+
+## Nexus
+
+* `nexus_user`
+* `nexus_pass`
+
+Provide a username and password to give your self-hosted Shields installation
+access to your private nexus repositories.
+
+## NPM
+
+* `npm_token`
+
+[Generate an npm token][npm token] to give your self-hosted Shields
+installation access to private npm packages
+
+[npm token]: https://docs.npmjs.com/getting-started/working_with_tokens
+
+## SymfonyInsight (formerly Sensiolabs)
+
+* `sl_insight_userUuid`
+* `sl_insight_apiToken`
+
+The SymfonyInsight API requires authentication. To obtain a token,
+Create an account, sign in and obtain a uuid and token from your
+[account page](https://insight.sensiolabs.com/account).
+
+## SonarQube
+
+* `sonarqube_token`
+
+Provide a token to give your self-hosted Shields installation access to a
+private SonarQube instance.
```

Comment:
I've just kind of guessed here. I'm not really sure. Whereas with the other services, I was able to work out the purpose of the credentials from the code or by going back over old issues/PRs, this one was more difficult. I'm not really clear on whether this allows us to access a private instance, or private projects on a public instance, or a bit of both. Does anyone know? Also, it appears there are a number of different permissions we can assign to a token: https://docs.sonarqube.org/display/DEV/Web+API#WebAPI-Noteaboutapi/permissions Do we know which are needed to generate a token for this purpose?

Reply:
I don't know definitively but I've hooked a self-hosted Shields instance up to a SonarQube server using this token so here's what I've seen:

My experience with creating tokens for Sonar (both for self-hosted SonarQube servers and SonarCloud) has always been the same, I just create a token and then use it elsewhere. The process described at https://docs.sonarqube.org/latest/user-guide/user-token/ has always been my experience. Sonar doesn't ask what permissions/scopes to apply and I've never even seen anywhere I can change it.

The SonarQube server I connected had all of our projects pubic, but it did require first authenticating. Therefore my guess is that the token is used for both scenarios you described above (both for authentication in general, and by sonar for authorization when requesting info from a private project)